### ISSUE TITLE
False positive on closing rows with external function/method

### ISSUE THREAD

**Issue by maksemen2** [ryanrolds/sqlclosecheck#43]
```
func deferRows(rows *sql.Rows) {
	err := rows.Close()
	if err != nil {
		log.Error(err)
	}
}

rows, err := db.QueryContext(ctx, "some query", arg1, arg2)
if err != nil {
	...
}
defer deferRows(rows)

```

will give us an error: ```error Rows/Stmt/NamedStmt was not closed (sqlclosecheck)```, although we are closing the rows with external function to avoid code duplicating.

**Comment by oxzi** [comment]
With a recent `golangci-lint` update, the bundled sqlclosecheck version was bumped to v0.6.0, resulting in quite similar false positives for us, also using `db.QueryContext`.

```go
        rows, err := db.QueryxContext(ctx, db.Rebind(stmt), args...)
        if err != nil {
                return err
        }
        defer func() { _ = rows.Close() }()

        for rows.Next() {
                row := new(Row)
                if err = rows.StructScan(row); err != nil {
                        return err
                }

                callback(row)
        }

        return rows.Err()
```

```
Rows/Stmt/NamedStmt was not closed (sqlclosecheck)
  rows, err := db.QueryxContext(ctx, db.Rebind(stmt), args...)
```

In contrast to what the similar issue #49 stated, calling `defer rows.Close()` directly instead of going through a function literal - as shown in the code above - does not change anything.
